pkg/chunk/table_manager.go

```diff
@@ -443,20 +443,28 @@ func (m *TableManager) partitionTables(ctx context.Context, descriptions []Table
 
 	if m.cfg.RetentionPeriod > 0 {
 		// Ensure we only delete tables which have a prefix managed by Cortex.
-		tablePrefixes := map[string]struct{}{}
+		tablePrefixes := map[string]*regexp.Regexp{}
 		for _, cfg := range m.schemaCfg.Configs {
 			if cfg.IndexTables.Prefix != "" {
-				tablePrefixes[cfg.IndexTables.Prefix] = struct{}{}
+				rx, err := regexp.Compile(fmt.Sprintf("%s\\d+$", cfg.IndexTables.Prefix))
```

Comment:
It's possible to have multiple entries in the schema table with the same prefix, so should check whether this prefix is in the map before compiling a new copy of the regex.

Reply:
`fmt.Sprintf("%s\\d+$", cfg.IndexTables.Prefix)` is a long way to write `cfg.IndexTables.Prefix + "\\d+$"`.

Reply:
You probably want `^` at the beginning too, otherwise you have replaced a prefix problem with a suffix problem.